tests/torchtune/modules/test_rms_norm.py

```diff
@@ -66,6 +64,7 @@ def test_forward_fp16(self, rms_norm, input_random_fp16, dim) -> None:
 
         # convert input to float since rms_norm computes in fp32
         expected_fp16 = normalize(input_random_fp16.float(), p=2, dim=-1) * (dim**0.5)
+        expected_fp16 = expected_fp16.to(torch.float16)
 
         assert_expected(output_fp16, expected_fp16, atol=1e-7, rtol=1e-3)
-        assert output_fp16.dtype == torch.float32
+        assert output_fp16.dtype == torch.float16
```

Comment:
do you know why this wasnt failing before?

Reply:
yeah the model wasn't casted to fp16, which means the scale parameter was still fp32. And since `x_normed * self.scale` occurred after the cast back to fp16, the output ended up in fp32.